src/ex4/server/services/item_manager.js

```diff
@@ -168,9 +208,20 @@ class ItemManager {
     return JSON.parse(cache);
   }
 
-  reSortTasks(newSortedTasks) {
-    this.tasks = newSortedTasks;
+  async reSortTasks(newSortedTasks) {
+    try {
+      for (let i = 0; i < newSortedTasks.length; i++) {
+        const task = newSortedTasks[i];
+        await Item.update({ position: i }, { where: { id: task.id } });
+      }
+      return true;
+    } catch (err) {
+      console.log(err);
+      return false;
+    }
   }
 }
 
-module.exports = new ItemManager();
+const itemManager = new ItemManager();
+itemManager.getTasks();
```

Comment:
this should be done on damand from the outside and not when requiring this class